### ISSUE TITLE
Full-page translation not working

### ISSUE THREAD

**Issue by vcekron** [MrAlex94/Waterfox#1932]
Hi all,

I noticed in the latest blog post and via your pull requests that you updated the Bing translation API, but I can't get it to work properly.

**Initial setup:**

1. Go to `about:config`.
2. Flip `browser.translation.detectLanguage` and `browser.translation.ui.show` to `true`.
3. ??? I'm not sure if we need to do anything else as the blog post just mentions "You can manually add a key to enable this feature in `about:config`"

**Steps to reproduce:**
1. Go to a foreign site, e.g., https://www.omroepbrabant.nl/
2. Click "Translate" in the UI that shows up

**Expected results:**
* The page is translated

**Observed results:**
* A message saying "There has been an error translating this page"
![scrShot](https://user-images.githubusercontent.com/1002359/106787781-69994000-6650-11eb-89ce-3ac042d3d503.png)

**Extra details:**
* I noticed that in `about:config` the `browser.translation.engine` flag is set to `Google`. I tried changing this to `Bing` but it changed nothing.
* ~~Some languages appear not to be identified. Try, e.g., https://www.sweclockers.com/. I don't get a popup here (I can open a separate issue about this later, if you prefer).~~ I guess this is because of this: https://wiki.mozilla.org/QA/Translation#Intro

Thanks in advance!

EDIT:
* I also tried this: https://wiki.mozilla.org/QA/Translation#Requirements with no success.
* I also tried this: https://www.reddit.com/r/openSUSE/comments/gmxdaj/firefox_translator_like_in_chrome/, but Yandex stopped offering free API keys.

**Comment by MrAlex94** [comment]
You need to create the relevant Microsoft Translate API keys in `about:config`:

`browser.translation.bing.clientIdOverride` with your Microsoft Translate API client ID.
`browser.translation.bing.apiKeyOverride` with your Microsoft Translate API key.

**Comment by vcekron** [comment]
Thanks, @MrAlex94, but I can't get it to work.

Looking at https://github.com/MrAlex94/Waterfox/blob/current/browser/components/translation/BingTranslator.jsm, I noticed that only `browser.translation.bing.apiKeyOverride` is used, and in particular that it needs to be in the `westeurope` region.

So I created the necessary API keys, and indeed using 
```
// Pass secret key using header
curl --header 'Ocp-Apim-Subscription-Key: <your-key>' --data "" https://westeurope.api.cognitive.microsoft.com/sts/v1.0/issueToken'

// Pass secret key using query string parameter
curl --data "" 'https://westeurope.api.cognitive.microsoft.com/sts/v1.0/issueToken?Subscription-Key=<your-key>'
```
from https://docs.microsoft.com/en-us/azure/cognitive-services/translator/reference/v3-0-reference#authenticating-with-an-access-token returns a token.

I still get an error when I try to translate the page, however, and I can't for the life of me figure out why. I don't know how to get a log or anything from WF, but I'll gladly do so if you have some further insight into how I can debug this.

Furthermore, I'll gladly write up a wiki page about setting this all up once we have it working, as it is an incredibly important aspect of a browser for me.
